### DISCUSSION TITLE
bids() function for filenames that are not "subject" based?

### DISCUSSION THREAD

**Discussion by akhanf** [khanlab/snakebids#204] [General]
There are a number of scenarios where you may not have subject-specific file naming, but still want to have BIDS-style names, and have another entity effectively take the place of `subject`.

e.g. for templates (`tpl-{template}`)
(see here for examples: https://www.templateflow.org/browse/)

or for aggregate data. 

In existing workflows I have used the `prefix=` keyword in `bids()` to emulate this, but it is a bit hacky as you would have to include something like: `prefix=f'tpl-{template}/tpl-{template}_'`

Any thoughts on (ideally non-breaking) changes that we could introduce to accommodate this?

**Comment by akhanf** [comment]
btw, other workarounds I have used in the past include using things like `sub-group` to indicate aggregated data, but that is also not ideal..

**Comment by pvandyken** [comment]
Is the limitation here that you need your non-standard entity to be a directory name as well as in the file name?

**Comment by akhanf** [comment]
Yes that's one limitation, I guess another is similar to something you brought up earlier that was discussed re: relative placement of the entities -- in that context it was to ensure `desc-` was always at the end, and here it would be to make sure `tpl-` was always at the beginning..

**Reply by pvandyken** [reply]
I wonder if all these concerns could be addressed by making some sort of formalized configuration document defining the bids template. This could be fed to `bids()` via some `format_file=...` parameter. Then the idea in production would be to override `bids()` at the beginning of the snakefile (e.g. `bids = ft.partial(bids, format_file=...)`).

This would make such changes accessible, while keeping them in the "advanced tier", as modifying the bids spec should be. And it would be more extensible for future similar requests)

**Reply by akhanf** [reply]
Yes something like this could work.. Having some canned configurations that are more easily accessible (ie that don't require a custom user config) for naming that is fairly commonplace (like tpl-{template}) would be helpful too..

**Reply by pvandyken** [reply]
That could be arranged fairly easily (e.g. passing a string vs a path, as in pybids config)

**Reply by tkkuehn** [reply]
Note: Current proposal for this is here for reference: https://github.com/akhanf/snakebids/issues/245
